ext/NormalizingFlowsCUDAExt.jl

```diff
@@ -0,0 +1,40 @@
+module NormalizingFlowsCUDAExt
+
+using CUDA
+using NormalizingFlows: Random, Distributions
+
+# Make allocation of output array live on GPU.
+function Distributions.rand(
+    rng::CUDA.RNG,
+    s::Distributions.Sampleable{<:Distributions.ArrayLikeVariate,Distributions.Continuous},
+)
+    return @inbounds Distributions.rand!(
+        rng, Distributions.sampler(s), CuArray{float(eltype(s))}(undef, size(s))
+    )
+end
+
+function Distributions.rand(
+    rng::CUDA.RNG,
+    s::Distributions.Sampleable{<:Distributions.ArrayLikeVariate,Distributions.Continuous},
+    n::Int,
+)
+    return @inbounds Distributions.rand!(
+        rng, Distributions.sampler(s), CuArray{float(eltype(s))}(undef, length(s), n)
+    )
+end
+
+function Distributions._rand!(rng::CUDA.RNG, d::Distributions.MvNormal, x::CuVecOrMat)
+    # Replaced usage of scalar indexing.
+    CUDA.randn!(rng, x)
```

Comment:
@zuhengxu do you know why this change of yours was necessary? I thought `Random.randn!(rng, x)` should just dispatch to `CUDA.randn!(rng, x)`?

Reply:
ahh, you are right---this is not necessary. I think I just made the change to ensure it's actually calling the cuda sampling. I can change it back.